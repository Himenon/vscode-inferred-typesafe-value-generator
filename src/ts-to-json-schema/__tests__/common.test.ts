import * as path from "path";
import * as Module from "../index";

const basePath = path.join(__dirname, "./sample-code/");

describe("Popular JSON Schema Test", () => {
  test("Load files directly", () => {
    const schema = Module.generateJsonSchema({
      files: [path.join(basePath, "./types.ts")],
      typeName: {
        isUnknown: false,
        firstTypeName: "ServerSideConfig",
        isArray: false,
        intersectinoTypeNames: [],
        isIntersectionType: false,
      },
      basePath,
    });
    expect(schema).toMatchSnapshot();
  });

  test("Reading files indirectly", () => {
    const schema = Module.generateJsonSchema({
      files: [path.join(basePath, "./sample.ts")],
      typeName: {
        isUnknown: false,
        firstTypeName: "ServerSideConfig",
        isArray: false,
        intersectinoTypeNames: [],
        isIntersectionType: false,
      },
      basePath,
    });
    expect(schema).toMatchSnapshot();
  });
});
