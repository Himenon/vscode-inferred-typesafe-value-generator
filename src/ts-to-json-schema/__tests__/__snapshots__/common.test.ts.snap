// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular JSON Schema Test Load files directly 1`] = `
Object {
  "jsonSchema": Object {
    "$ref": "#/definitions/ServerSideConfig",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": Object {
      "ServerSideConfig": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
          "service_a": Object {
            "additionalProperties": false,
            "properties": Object {
              "url": Object {
                "type": "string",
              },
            },
            "required": Array [
              "url",
            ],
            "type": "object",
          },
          "service_b": Object {
            "additionalProperties": false,
            "properties": Object {
              "apiToken": Object {
                "type": "string",
              },
              "timeout": Object {
                "type": "number",
              },
              "url": Object {
                "anyOf": Array [
                  Object {
                    "const": "https://dev.example.com",
                    "type": "string",
                  },
                  Object {
                    "const": "https://prod.example.com/b",
                    "type": "string",
                  },
                ],
              },
            },
            "required": Array [
              "url",
              "timeout",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "service_b",
        ],
        "type": "object",
      },
    },
  },
}
`;

exports[`Popular JSON Schema Test Reading files indirectly 1`] = `
Object {
  "jsonSchema": Object {
    "$ref": "#/definitions/ServerSideConfig",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": Object {
      "ServerSideConfig": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
          "service_a": Object {
            "additionalProperties": false,
            "properties": Object {
              "url": Object {
                "type": "string",
              },
            },
            "required": Array [
              "url",
            ],
            "type": "object",
          },
          "service_b": Object {
            "additionalProperties": false,
            "properties": Object {
              "apiToken": Object {
                "type": "string",
              },
              "timeout": Object {
                "type": "number",
              },
              "url": Object {
                "anyOf": Array [
                  Object {
                    "const": "https://dev.example.com",
                    "type": "string",
                  },
                  Object {
                    "const": "https://prod.example.com/b",
                    "type": "string",
                  },
                ],
              },
            },
            "required": Array [
              "url",
              "timeout",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "service_b",
        ],
        "type": "object",
      },
    },
  },
}
`;
